transcom/mymove
# B 18165 - Add TOO edits of PPM details to Move history

## [B-18165](https://www13.v1host.com/USTRANSCOM38/story.mvc/Summary?oidToken=Story%3a871851)

## Summary

This PR addresses logging the actions of TOO users in our Move history when they make edits to PPM details. Most of the code changes are standard for adding an event to the Move history of a move, but there are a few details to note:

- The trigger added to ppm_shipments covers changes to ~40 columns. Just something to keep in mind.
- I touched two files that were semi out of scope of the direct AC of this backlog item
    1. I updated formatCustomerDate in formatters.js to return our DEFAULT_EMPTY_VALUE if !date, to avoid nulls resulting in "Invalid date"
    2. I added a booleanFields.js file that keeps the design pattern of constants for our different Label fields (dateFields, weightFields) and updated the LabeledDetails.jsx to reflect that
- I also updated our field mappings to add the word weight when dealing with Pro-gear weight or Spouse pro-gear weight (and updated all affected test files) to avoid confusion in these scenarios:
![image](https://github.com/transcom/mymove/assets/147535220/20cacbc8-634a-4748-be9b-833520066d87)


## Verification Steps for the Author

These are to be checked by the author.


## Verification Steps for Reviewers

These are to be checked by a reviewer.


### How to test

1. Login as a TOO and find a move that has a PPM shipment
2. Make edits to the PPM shipment, save, and make sure those edits are reflected when you view the Move history tab
